advanced/modules/README.md

```diff
@@ -0,0 +1,127 @@
+will be recreated if the source code changes. Moreover, if you want to slightly modify one of the instances, you have to create two mostly
+identical Lua source files, and thus create code duplication.
+
+Modules provide a solution to this problem which works very similar to how typical libraries or modules in other languages work.
+You can put some of the functions and data you need in your project in a module, and reference that module from multiple scripts.
+Let's build an example which does that!
+
+## Copy pasta
+
+In this tutorial, we will use the [monkey example](../../basics/monkey/README.md#recreate-sample) and build on top of it.
+In that example, we created three monkey heads which are lit by a Phong shader which is in turn controlled by a Lua script.
+Let's imagine we wanted to create a second, different lighting model, but we wanted to keep the light configuration and the color
+settings shared between the old and the new shading model.
+
+We notice that that the basic monkey lighting script has default values for the light properties, which are set in the GLOBAL table
+during initialization of the script. Also, the run() function checks if a non-zero value is set for the colors, and if not, uses
+the default colors. We want to move both the colors and the function which decides whether to use a default color or not into a new module.
+
+## Creating the light module
+
+To create a new module, we need to create a new Lua source file and write its code there. We follow the
+[official Logic Engine docs](https://ramses-logic.readthedocs.io/en/latest/lua_syntax.html#custom-modules) and
+create the following [Lua module source file](./scripts/LightModule.lua). You may notice that the file looks quite
+similar to the LightControl script from the basic monkey example, with a few structural differences:
+
+* The module has no global functions such as interface()
+* Instead, it declares a table (called light) in the beginning of the file and returns it to the caller at the end
+* In the table, the module puts data (color and light properties) and functions to obtain those (getLightDirection(), resolveColor())
+* It has more error checks than the original LightControl script
+
+We will explain later why this is necessary. Let's move on and use the new module first!
+
+## Using the module
+
+Modules have to be explicitly created and imported in the scripts which need to access them. To create a module,
+right click the resources area and create a new object of type 'LuaScriptModule'. Set its URI to the source code file
+we created in the previous section.
+
+To let a script make use of a module, you have to explicitly declare you want to use it. Add the following line in the
+LightControl script:
+
+```lua
+modules('light')
+
+function interface()
+...
+
+```
+
+Now the LightControl script will report errors, because you declared a module but didn't provide one to the script. To fix the
```

Comment:
```suggestion
Now the LightControl script will report errors, because you declared a module in the script file but didn't provide a module to the respective LuaScript object in Ramses Composer. To fix the
```